master/internal/api_token.go

```diff
@@ -36,18 +37,37 @@ func (a *apiServer) PostAccessToken(
 		return nil, status.Error(codes.PermissionDenied, err.Error())
 	}
 
-	tokenExpiration := token.DefaultTokenLifespan
-	if req.Lifespan != "" {
-		d, err := time.ParseDuration(req.Lifespan)
-		if err != nil || d < 0 {
-			return nil, status.Error(codes.InvalidArgument,
-				"Lifespan must be a Go-formatted duration string with a positive value")
+	defaultTokenLifespan, err := token.ParseLifespanDays(a.m.config.Security.Token.DefaultLifespanDays)
+	if err != nil {
+		return nil, status.Error(codes.Internal, "Failed to parse default lifespan")
+	}
+
+	maxTokenLifespan, err := token.ParseLifespanDays(a.m.config.Security.Token.MaxLifespanDays)
+	if err != nil {
+		return nil, status.Error(codes.Internal, "Failed to parse max lifespan")
+	}
+
+	tokenExpiration := defaultTokenLifespan
+	if req.Lifespan != nil {
+		if *req.Lifespan == config.InfiniteTokenLifespanString {
+			tokenExpiration = maxTokenLifespan
+		} else {
+			d, err := time.ParseDuration(*req.Lifespan)
+			if err != nil || d < 0 {
```

Comment:
let's differentiate these cases and return the error if there is one. it's probably got useful debugging information for the user.

Reply:
(we can wrap it, but as is the err is discarded)

Reply:
separated both the checks, and logged them